gq/gq_test.go

```diff
@@ -0,0 +1,86 @@
+package gq_test
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/rsa"
+	"crypto/sha256"
+	"encoding/json"
+	"io"
+	"testing"
+	"time"
+
+	insecurerand "math/rand"
```

Comment:
Could you use: `"crypto/rand"` here instead or is there a dependency on the RSA library?

See example of crypto/rand usage:
https://github.com/bastionzero/openpubkey/blob/main/pktoken/signer.go#L186

Reply:
Apologizes, that's my mistake. I thought that was the RNG for gq, not the test, the file name `gq_test.go` should have tipped me off.

> I realize I've only half-implemented what I was planning though - to seed the rng with a random number which we also print out, so that if there's some failure in a test to do with the random input we can reproduce it easily.

That's an excellent idea. If you figure out a way to implement it with rsa.GenerateKey, we should do it. If not, don't worry about it.